projects/canopy/src/lib/card/card.component.ts

```diff
@@ -32,19 +36,40 @@ let nextUniqueId = 0;
 export class LgCardComponent implements AfterContentInit, OnDestroy {
   private subscription: Subscription;
   private uniqueId = nextUniqueId++;
+  orientationClasses: Array<string> = [];
 
   @ContentChild(forwardRef(() => LgButtonToggleDirective))
   buttonToggle: LgButtonToggleDirective;
   @ContentChild(forwardRef(() => LgCardToggableContentComponent))
   cardToggableContent: LgCardToggableContentComponent;
   @ContentChild(forwardRef(() => LgCardNavigationTitleComponent))
   cardNavigationTitle: LgCardNavigationTitleComponent;
+  @Input() variant: CardVariant = 'default';
 
-  constructor(private renderer: Renderer2, private hostElement: ElementRef) {}
+  constructor(
+    private renderer: Renderer2,
+    private hostElement: ElementRef,
+    private spacingService: SpacingService,
+  ) {}
+
+  @Input()
+  set orientation(orientation: SpacingResponsive) {
+    this.orientationClasses = this.toggleClasses(
+      this.spacingService.createNewClasses(orientation, 'orientation'),
```

Comment:
As `SpacingService` is now being used for more than just the spacing module, I think it's time to make it more generic. Perhaps we can rename it to `StylingService` and move it to _src/lib/utils_. This would require some re-testing of the Spacing directives, but noting that there is an existing bug with the responsive styling: https://github.com/Legal-and-General/canopy/issues/1091 which of course won't be fixed with this change.

Another thing we need to consider is that the `createNewClasses` method also calls the `DynamicStyleService` to dynamically add CSS code to the <head> of the web page - this was a way of avoiding having tons and tons of CSS to download just for responsive spacing. However, I don't think we need it here for `orientation`, and it would be quite involved to make it work anyway. Right now, this method will be trying to create these styles but they are probably a bit weird looking! Perhaps we can add another param to `createNewClasses` which lets you optionally add the dynamic styles.

For example... (note the comment and param changes)

```
  /**
   * Returns the correct class names based on the rules provided
   * e.g. ['lg-margin__right'], or ['.lg-orientation--sm--horizontal']
   * Optional: Creates new CSS selector and declaration based on the spacing value
   * which are applied via the DynamicStyleService:
   * e.g. `.lg-padding`, `padding: 1rem`
   *
   * @param rules e.g. 'sm' for SpacingVariant, or {sm: 'lg', md: 'xxl'} for SpacingResponsive, or { sm: 'vertical', md: 'horizontal'} for OrientationResponsive
   * @param cssProperty e.g. 'padding-top' or 'margin-right'
   * @returns an array of the created class names
   */
createNewClasses(
    rules: SpacingVariant | SpacingResponsive | OrientationResponsive, // renamed from spacing to rules
    cssProperty: string, 
    dynamicallyAddSpacingStyles = false // used in spacing module only
  ): Array<string> {

// usage for Orientation
this.stylingService.createNewClasses(orientation, 'orientation'),

// usage for Spacing
this.stylingService.createNewClasses(spacing, 'padding-top', true),
```

Probably a good idea if we have a call to discuss :)

Reply:
Alternatively, we could opt not to reuse `SpacingService`, and create a smaller method on `LgCardComponent`. Something like this:

```
createNewOrientationClasses(
    rules: OrientationResponsive,
  ): Array<string> {
    const newClasses = [];

    Object.keys(rules).forEach(key => {
      newClasses.push(`lg-orientation--${key}--${rules[key]}`);
    });

    return newClasses;
  }
```

Thinking about it, maybe this is better - trying to repurpose `SpacingService` might be adding too much complexity, as it's pretty tied up with the spacing module implementation. What do you think? Perhaps we can go with this simpler approach and add an issue to look at consolidating/generalising later - could also look at dynamically adding those orientation styles too at that point to reduce the published bundle size.

Reply:
Done. Created a smaller method on LgCardComponent for this :)